app/src/main/kotlin/com/wire/android/ui/home/settings/account/displayname/ChangeDisplayNameViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package com.wire.android.ui.home.settings.account.displayname
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.navigation.EXTRA_SETTINGS_DISPLAY_NAME_CHANGED
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.kalium.logic.feature.user.DisplayNameUpdateResult
+import com.wire.kalium.logic.feature.user.GetSelfUserUseCase
+import com.wire.kalium.logic.feature.user.UpdateDisplayNameUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ChangeDisplayNameViewModel @Inject constructor(
+    private val getSelf: GetSelfUserUseCase,
+    private val updateDisplayName: UpdateDisplayNameUseCase,
+    private val navigationManager: NavigationManager,
+    private val dispatchers: DispatcherProvider
+) : ViewModel() {
```

Comment:
I know this VM handles mostly UI, but since we are dropping a 0.14% on code coverage can we at least have some easy unit test class covering at least the use case calls to cover error responses? Extra points if we are also asserting the correct error is displayed via the final status of `displayNameState` 😉